tests/test_recorddumploader.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Module tests."""
+
+from __future__ import absolute_import, print_function
+
+from datetime import datetime
+
+import pytest
+from invenio_files_rest.models import Bucket, BucketTag, FileInstance, \
+    ObjectVersion
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus, \
+    RecordIdentifier
+from invenio_records.api import Record
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_migrator.records import RecordDumpLoader
+
+
+def test_new_record(app, db, dummy_location, record_dump, resolver):
+    """Test creation of new record."""
+    RecordDumpLoader.create(record_dump)
+    pid, record = resolver.resolve('11782')
+    # Basic some test that record exists
+    assert record['title']
+    assert record.created == datetime(2014, 9, 19, 14, 6, 50)
+    # Test that this is a completely new record
+    assert len(record.revisions) == 2
+    pytest.raises(IntegrityError, RecordIdentifier.insert, 11782)
+    # Test the PIDs are extracted and created
+    assert PersistentIdentifier.get('doi', '10.5281/zenodo.11782')
+
+    assert len(record['files']) == 1
+    f = record['files'][0]
+    obj = ObjectVersion.get(f['bucket'], f['filename'])
+    assert obj.file.checksum == f['checksum']
```

Comment:
The RecordDumpLoader will blindly take the checksum from the legacy dump and create a file instance with the same checksum (i.e. dump checksum (``f['checksum'])``) should be the same as the object file instance checksum (``obj.file.checksum``).

If you have no files in the legacy dump, no objects/file instances are created. In this case, checksum is calculated in the normal use of Files-REST (either calculated on ``set_contents``, or you provide it).

Note, we don't check fixity during the migration, but it's assumed that after the migration you will have to run the fixity checker provided by Invenio-Files-REST for all files.